### DISCUSSION TITLE
load alpinejs using importmap

### DISCUSSION THREAD

**Discussion by kirantpatil** [alpinejs/alpine#3077] [1. Help] [answered]
Hi,

Rails 7 by default uses [importmap](https://github.com/rails/importmap-rails) to load JS.

Please provide the steps to load alpinejs to Rails 7 using importmap.

Regards,
Kiran.

**Comment by kirantpatil** [comment, marked as answer]
This command helped me.
$ ./bin/importmap pin alpinejs --download

**Reply by hooopo** [reply]
Couldn't find any packages in ["alpinejs", "--download"] on jspm

**Comment by leoplct** [comment]
I have the same error. I

**config/importmap.rb**
`pin "alpinejs" # @3.13.3`


I got this error
```
Asset `alpinejs/dist/module.cjs.js` was not declared to be precompiled in production.
Declare links to your assets in `app/assets/config/manifest.js`.

  //= link alpinejs/dist/module.cjs.js
```

**Comment by ekwoka** [comment]
```json
{
  "imports": {
    "alpinejs": "https://esm.sh/alpinejs"
  }
}
```
